### ISSUE TITLE
ads in New UI

### ISSUE THREAD

**Issue by amd64fox** [mrpond/BlockTheSpot#150]
Good afternoon, I noticed that now supports the new Alpha version (New UI) appeared in the [features](https://github.com/mrpond/BlockTheSpot#features)  description.

I still see an empty ad block, am I the only one?

![Бsdsadsaез имени](https://user-images.githubusercontent.com/62529699/115571971-28681200-a2c8-11eb-9bdb-3b42f355b558.jpg)

![Без им2321ени](https://user-images.githubusercontent.com/62529699/115572142-4c2b5800-a2c8-11eb-9937-fdcff064105e.jpg)

**Comment by ronny941** [comment]
yes it is true there is the bar where the advertisement should appear but although there is it is blocked thanks to blockthespot and the songs work I hope you can remove it completely but everything works and nothing has appeared

**Comment by ryzenblue** [comment]
I have a workaround that works for now.
1. Navigate to `%APPDATA%/Spotify/Apps`. Here you will find xpui.spa, make a copy of this file so that you have a backup.
2. Rename xpui.spa to xpui.zip and open the zip. Within the zip, there will be xpui.css, extract this file.
3. Open the extracted file with a text editor and search for `.b2731cdeb97c1f6a4cf5caa44e943acb-scss`. This is the css selector of the ad banner div.
4. For this selector, change the `display:flex` to `display:none`, save the file, **drop it back in the zip to replace** the original xpui.css.
5. Change the extension from xpui.zip to xpui.spa

The div selector did not change even after I reinstalled, so I'm guessing it should be the same for everybody.

**Comment by ronny941** [comment]
this fix will it be added to blockthespot?
thank you in advance

**Comment by rednek46** [comment]
@ronny941 if those files are not changed everytime after a new update, it will be added.

**Comment by ronny941** [comment]
> 
> 
> @ronny941 if those files are not changed everytime after a new update, it will be added.

the new update of spotify is now out 
but I can't tell you if the fix works completely you wait the user who gave you this fix

**Comment by amd64fox** [comment]
> the new update of spotify is now out
> but I can't tell you if the fix works completely you wait the user who gave you this fix

files change, so this method is unlikely to be added to BTS

**Comment by ronny941** [comment]
> 
> 
> > the new update of spotify is now out
> > but I can't tell you if the fix works completely you wait the user who gave you this fix
> 
> files change, so this method is unlikely to be added to BTS

ok i understand so you have to edit that file every time
because with every update of spotify the files change because it puts its originals back
just like when blockthespot has to be reapplied for the same reason

**Comment by amd64fox** [comment]
Maybe it will help yet, in the `xpui.js` file you need to find this `e=>e.ads.leaderboard.` and replace `Enabled` with `Disabled`, after which the empty block disappears. Also in this file, I was able to find a line to remove the `"Upgrade"` button, but as I understand it, the problem above persists and this method cannot be added to BTS.

**Comment by ronny941** [comment]
> 
> 
> Maybe it will help yet, in the `xpui.js` file you need to find this `e=>e.ads.leaderboard.` and replace `Enabled` with `Disabled`, after which the empty block disappears. Also in this file, I was able to find a line to remove the `"Upgrade"` button, but as I understand it, the problem above persists and this method cannot be added to BTS.

I can understand it because the files change with every spotify update and therefore it is impossible to always add these changes to blockthespot

**Comment by Kibakus** [comment]
Hello everyone, maybe my script on Python will help someone. He removes the button and "Empty AD Block".
[Not advertising](https://github.com/Kibakus/SpotifyBlockNewUI)

**Comment by Sruko** [comment]
Use `Block_BannerOnly = 0` if you use 1 you will see ad banner placeholder because patch is not being applied.

**Comment by rednek46** [comment]
@Kibakus good job on the python script. 
But I don't think it will be added to BTS. Because, BTS already blocks the AD Block. Its just only the New Alpha UI That has the bug. Which will be fixed eventually.
-----------------------

@Sruko `Block_BannerOnly` is only for Spotify Premium Users.

**Comment by Sruko** [comment]
> @Kibakus good job on the python script.
> But I don't think it will be added to BTS. Because, BTS already blocks the `Black AD Block element`. Its just only the New Alpha UI That has the bug. Which will be fixed eventually.
> 
> @Sruko `Block_BannerOnly` is only for Spotify Premium Users.

That doesn't matter at all, if you enable it the banner code is not run even if you are not premium. You need to disable it or you will continue to see ads.

**Comment by rednek46** [comment]
@Sruko BTS normally removes the banner for free users without editing the config. I think in the new UI banner is not being blocked correctly.

**Comment by Sruko** [comment]
> @Sruko BTS normally removes the banner for free users without editing the config. I think in the new UI banner is not being blocked correctly.

You can easily test it yourself, if the code as is isn't already clear. If the option is enabled the process will not get patched with the banner removal, and thus you will continue to see it + ads. The current version of BTS already works on new Spotify.

**Comment by RichardLeblanc** [comment]
> 
> 
> Maybe it will help yet, in the `xpui.js` file you need to find this `e=>e.ads.leaderboard.` and replace `Enabled` with `Disabled`, after which the empty block disappears. Also in this file, I was able to find a line to remove the `"Upgrade"` button, but as I understand it, the problem above persists and this method cannot be added to BTS.

Can you telll which is the line in this file to remove the upgrade button?
Thanks in advance

**Comment by amd64fox** [comment]
@RichardLeblanc 
need to find and delete this `createElement(pu,{onClick:C,className:gu.Z.UpgradeButton}),i().`

**Comment by Sruko** [comment]
Ok my bad, seems there was some inconsistency on my side in terms of Spotify rollout, now the banner actually shows but ads are blocked, the current workaround editing xpui to hide it and patching Spotify directly ("00 83 C4 08 84 C0 0F 84 E5" to "00 83 C4 08 84 E4 0F 84 E5") or via hook to block ads work.

**Comment by Sruko** [comment]
Also it seems you can block ads/remove upgrade button/remove premium text in menu just by editing a line in xpui.js.
`{var t,n,r;return"free"===(null===(t=e.session)||void 0===t||null===(n=t.productState)||void 0===n||null===(r=n.catalogue)||void 0===r?void 0:r.toLowerCase())},i=e=>{var t,n,r;return"premium"===(null===(t=e.session)||void 0===t||null===(n=t.productState)||void 0===n||null===(r=n.catalogue)||void 0===r?void 0:r.toLowerCase())}` 
to
`{var t,n,r;return},i=e=>{var t,n,r;return"premium"}`
![image](https://user-images.githubusercontent.com/83501246/116917274-bb049b80-ac4e-11eb-922f-961d33ffeab4.png)

The leaderboard can also be removed this as alternative
`adsEnabled:!0` to `adsEnabled:!1`

**Comment by rednek46** [comment]
Please respect the issue title before posting a comment. Create issue if yours is new or post them on their respective open issues.
Please don't try to deviate from the title.

Upgrade button issues/patches goes to #83. The black ad placeholder is a known bug and will be fixed in time. 
For further discussions related to this thread other than possible fixes not mentioned here, use discord.

**Comment by mrpond** [comment]
> Ok my bad, seems there was some inconsistency on my side in terms of Spotify rollout, now the banner actually shows but ads are blocked, the current workaround editing xpui to hide it and patching Spotify directly ("00 83 C4 08 84 C0 0F 84 E5" to "00 83 C4 08 84 E4 0F 84 E5") or via hook to block ads work.

I had hard time finding this, wow. thank for this. I will test it and will patch it into source if this work !!!

good job!

**Comment by mrpond** [comment]
@Sruko ("00 83 C4 08 84 C0 0F 84 E5" to "00 83 C4 08 84 E4 0F 84 E5")
I can't find it, where do I suppose to look at?

**Comment by Sruko** [comment]
This is just your existing patch for Alpha with a small change, it only blocks ads but will not remove the new empty banner at the bottom because it's now handled different.
What i changed is that instead of jumping, it changes to 'test AH, AH' instead of AL, AL, it's a simpler one byte patch.

To patch out the new banner from Spotifty.exe in Alpha is going to be a lot more involved tracing down.

**Comment by mrpond** [comment]
that right more tracing down. 
I don't want to patch 'libcef.dll' yet

**Comment by SaberAkiyama** [comment]
Hi. I have some a problem. I use BlockTheSpot with [Spicetify-CLI](https://github.com/khanhas/spicetify-cli) it's a custom theme version of Spotify, but the ads is popped up. Because version 2.1.0 in Spicetify-CLI has change XPUI script. I'd try to replace BlockTheSpot in the latest release, but it didn't work properly.

**Comment by Daksh777** [comment]
I managed to remove the black placeholder and also the "Upgrade" button from the new UI using Spicetify.
Go ahead and try it out: https://github.com/Daksh777/SpotifyNoPremium

**Comment by Rednek46** [comment]
@Daksh777 nice. I was able to block the placeholder and upgrade button. but couldn't figure out how to remove the upgrade from the profile drop down menu.

**Comment by fosam** [comment]
> I managed to remove the black placeholder and also the "Upgrade" button from the new UI using Spicetify.
> Go ahead and try it out: https://github.com/Daksh777/SpotifyNoPremium

@Daksh777 How can we do this with "SpicetifyDefault" Theme v2.1.0?

**Comment by Daksh777** [comment]
> @Daksh777 How can we do this with "SpicetifyDefault" Theme v2.1.0?

It's pretty easy, you can copy everything in user.css in my repo after the "For new UI" line and paste it into Spicetify Default's user.css

**Comment by fosam** [comment]
Ok, i'll report back here

**Comment by Daksh777** [comment]
> And before applying the theme right?

What do you mean?

**Comment by fosam** [comment]
i was asking if i have to copy the lines from your css before i apply the theme

**Comment by fosam** [comment]
I copied the lines from your css and then i applied the customization using spicetify, but there is a color (light brown) added to the theme that does not match the original design of the theme, maybe i did something wrong?

I deleted everything from Spicetify Default's user.css before pasting in the lines from your css, is this the correct way?

**Comment by Daksh777** [comment]
> I copied the lines from your css and then i applied the customization using spicetify, but there is a color (light brown) added to the theme that does not match the original design of the theme, maybe i did something wrong?
> 
> I deleted everything from Spicetify Default's user.css before pasting in the lines from your css, is this the correct way?

No, you had to just add all the lines after "For new UI" line in my user.css in Spicetify Default's user.css, you didn't have to delete it all.

**Comment by fosam** [comment]
You are right, Daksh.

Now i did it the correct way and it works as expected.

Question: did you manage to make that code yourself? because i see that it is different than the solutions some users provided above

**Comment by SaberAkiyama** [comment]
Try this one:
```spicetify config inject_css 0 replace_colors 0```

**Comment by Daksh777** [comment]
> Question: did you manage to make that code yourself? because i see that it is different than the solutions some users provided above

Yes, it's quite simple. All you need is to enable developer tools in Spotify and know CSS.

**Comment by fosam** [comment]
@Daksh777 Your workaround is highly recommended it works perfectly.

**Comment by fosam** [comment]
> Try this one:
> `spicetify config inject_css 0 replace_colors 0`

What does it do?

**Comment by SaberAkiyama** [comment]
> > Try this one:
> > `spicetify config inject_css 0 replace_colors 0`
> 
> What does it do?

It means back to original theme of Spotify

**Comment by fosam** [comment]
@SaberAkiyama it is an alternative to `spicetify restore`

Thank you!

**Comment by SaberAkiyama** [comment]
> @SaberAkiyama it is an alternative to `spicetify restore`
> 
> Thank you!

That was restore a backup to original files, it means Spotify Apps folders into ```.spa``` files

**Comment by fosam** [comment]
Ah Ok, so `spicetify config inject_css 0 replace_colors 0` restores only the <ins>original Spotify theme</ins> without restoring Spotify to original state?

**Comment by SaberAkiyama** [comment]
> Ah Ok, so `spicetify config inject_css 0 replace_colors 0` restores only the original Spotify theme without restoring Spotify to original state?

Yes.

**Comment by Rednek46** [comment]
@Daksh777 @fosam @SaberAkiyama please don't populate this thread from issues other than the Title.
Spicetify is a well documented repo, please read their docs and use their issue section to keep stuff clean and close to this thread. in case of further detailed discussion, you are open to use our discord.

**Comment by fosam** [comment]
@Rednek46 OK, I will keep the topic within the title of this issue

> @Daksh777 nice. I was able to block the placeholder and upgrade button. but couldn't figure out how to remove the upgrade from the profile drop down menu.

Have you managed to remove the "upgrade" from the profile drop down menu?

I don't have it there.

**Comment by SwissCheese99** [comment]
> 
> 
> I have a workaround that works for now.
> 
>     1. Navigate to `%APPDATA%/Spotify/Apps`. Here you will find xpui.spa, make a copy of this file so that you have a backup.
> 
>     2. Rename xpui.spa to xpui.zip and open the zip. Within the zip, there will be xpui.css, extract this file.
> 
>     3. Open the extracted file with a text editor and search for `.b2731cdeb97c1f6a4cf5caa44e943acb-scss`. This is the css selector of the ad banner div.
> 
>     4. For this selector, change the `display:flex` to `display:none`, save the file, **drop it back in the zip to replace** the original xpui.css.
> 
>     5. Change the extension from xpui.zip to xpui.spa
> 
> 
> The div selector did not change even after I reinstalled, so I'm guessing it should be the same for everybody.

Maybe a stupid question, but how do you change xpui.zip to xpui.spa at the end? When I first did spa->zip it didn't show zip anymore, so I can't really rename the file. Any ideas?

**Comment by AcelynnBen** [comment]
> I have a workaround that works for now.
> 
> 1. Navigate to `%APPDATA%/Spotify/Apps`. Here you will find xpui.spa, make a copy of this file so that you have a backup.
> 2. Rename xpui.spa to xpui.zip and open the zip. Within the zip, there will be xpui.css, extract this file.
> 3. Open the extracted file with a text editor and search for `.b2731cdeb97c1f6a4cf5caa44e943acb-scss`. This is the css selector of the ad banner div.
> 4. For this selector, change the `display:flex` to `display:none`, save the file, **drop it back in the zip to replace** the original xpui.css.
> 5. Change the extension from xpui.zip to xpui.spa
> 
> The div selector did not change even after I reinstalled, so I'm guessing it should be the same for everybody.

They removed this command so blocking it with this is no longer an option, either that or even the search in notepad ++  is blind lol

**Comment by Daksh777** [comment]
> They removed this command so blocking it with this is no longer an option, either that or even the search in notepad ++ is blind lol

Right, they changed the name of almost all classes. However, I have fixed this issue in https://github.com/Daksh777/SpotifyNoPremium

**Comment by Arrotanis** [comment]
> 
> 
> > I have a workaround that works for now.
> > 
> > 1. Navigate to `%APPDATA%/Spotify/Apps`. Here you will find xpui.spa, make a copy of this file so that you have a backup.
> > 2. Rename xpui.spa to xpui.zip and open the zip. Within the zip, there will be xpui.css, extract this file.
> > 3. Open the extracted file with a text editor and search for `.b2731cdeb97c1f6a4cf5caa44e943acb-scss`. This is the css selector of the ad banner div.
> > 4. For this selector, change the `display:flex` to `display:none`, save the file, **drop it back in the zip to replace** the original xpui.css.
> > 5. Change the extension from xpui.zip to xpui.spa
> > 
> > The div selector did not change even after I reinstalled, so I'm guessing it should be the same for everybody.
> 
> They removed this command so blocking it with this is no longer an option, either that or even the search in notepad ++ is blind lol

Just search for `height:106px` or just `106` (it's the only element with that height).

**Comment by amd64fox** [comment]
@Daksh777  @Arrotanis 
Why do you hide this banner through css, when lines have already been found in js that have not changed for a long time.
In my repository this is exactly what I did, https://github.com/amd64fox/SpotX

**Comment by noxan-dev** [comment]
> I have a workaround that works for now.
> 
> 1. Navigate to `%APPDATA%/Spotify/Apps`. Here you will find xpui.spa, make a copy of this file so that you have a backup.
> 2. Rename xpui.spa to xpui.zip and open the zip. Within the zip, there will be xpui.css, extract this file.
> 3. Open the extracted file with a text editor and search for `.b2731cdeb97c1f6a4cf5caa44e943acb-scss`. This is the css selector of the ad banner div.
> 4. For this selector, change the `display:flex` to `display:none`, save the file, **drop it back in the zip to replace** the original xpui.css.
> 5. Change the extension from xpui.zip to xpui.spa
> 
> The div selector did not change even after I reinstalled, so I'm guessing it should be the same for everybody.

The new class is .snMc3rl8H2pM9cqkTpQ6. Just follow the same instructions but instead of .b2731cdeb97c1f6a4cf5caa44e943acb use the new class

**Comment by Haider-kun** [comment]
Thanks @noxan-dev

**Comment by imloualvaro** [comment]
> > I have a workaround that works for now.
> > 
> > 1. Navigate to `%APPDATA%/Spotify/Apps`. Here you will find xpui.spa, make a copy of this file so that you have a backup.
> > 2. Rename xpui.spa to xpui.zip and open the zip. Within the zip, there will be xpui.css, extract this file.
> > 3. Open the extracted file with a text editor and search for `.b2731cdeb97c1f6a4cf5caa44e943acb-scss`. This is the css selector of the ad banner div.
> > 4. For this selector, change the `display:flex` to `display:none`, save the file, **drop it back in the zip to replace** the original xpui.css.
> > 5. Change the extension from xpui.zip to xpui.spa
> > 
> > The div selector did not change even after I reinstalled, so I'm guessing it should be the same for everybody.
> 
> The new class is .snMc3rl8H2pM9cqkTpQ6. Just follow the same instructions but instead of .b2731cdeb97c1f6a4cf5caa44e943acb use the new class

anyone knows a similar solution on mac osx?

**Comment by Araxeus** [comment]
The easiest way for me is editing xpui.js while keeping in mind that `x` can change between patches:
`function(x){return x.ads.leaderboard.isEnabled}` to `function(x){return x.ads.leaderboard.isDisabled}`

and for upgrade button:
DELETE `createElement(Xx,{onClick:x,className:Xx.x.UpgradeButton}),x().`

**Comment by NguyenASang** [comment]
Hi guys, I created a bat file to remove "Black Ad placeholder" .  When I finish the code and run it, no errors appear. But when I open Spotify it shows error: `error-dialog.generic.header`. Can anyone help me fix that :((
My code:
```
@echo off
powershell -Command "& {Invoke-WebRequest -UseBasicParsing 'https://raw.githubusercontent.com/mrpond/BlockTheSpot/master/install.ps1' | Invoke-Expression}"
cd %USERPROFILE%\AppData\Roaming\Spotify\Apps
ren xpui.spa xpui.zip
powershell -Command Expand-Archive -LiteralPath '%USERPROFILE%\AppData\Roaming\Spotify\Apps\xpui.zip' -DestinationPath '%USERPROFILE%\AppData\Roaming\Spotify\Apps\xpui'
del xpui.zip
cd xpui
powershell -Command "(gc xpui.css) -replace 'display:flex;height:106px', 'display:none;height:106px' | Out-File -encoding ASCII xpui.css"
cd ..
powershell -Command Compress-Archive -LiteralPath '%USERPROFILE%\AppData\Roaming\Spotify\Apps\xpui' -DestinationPath '%USERPROFILE%\AppData\Roaming\Spotify\Apps\xpui.zip'
ren xpui.zip xpui.spa
del xpui
exit
```

**Comment by Nuzair46** [comment]
The latest commit on the BlockTheSpot.bat will remove this placeholder.
Its an Experimental feature thanks to @compilerdeceiver397 .

**Comment by NguyenASang** [comment]
I created a bat file to remove that and it works fine, I intend to tell you but you did that :)))
Anyway thank you very much ❤️

**Comment by Deagan** [comment]
Was there a way I could auto-confirm the (Y/N) prompt, so that it removes the ad placeholder automatically? I've attempted to modify the .bat myself, without success. 😅

**Comment by noxan-dev** [comment]
> Was there a way I could auto-confirm the (Y/N) prompt, so that it removes the ad placeholder without having to confirm each start-up? I've attempted to do so through the .bat myself, without success. 😅

if you download the latest .bat this prompt was added.

**Comment by Nuzair46** [comment]
@Deagan You only have to use the bat file once. Then you can normally start Spotify.

**Comment by voltrare** [comment]
Im not using the script because im installing with scoop
I don't know if its only me but changing the xpui.js makes spotify slow/ not responsive (by directly editing xpui.js)
BUT changing the only the css does not give that problem

**Comment by Araxeus** [comment]
> I don't know if its only me but changing the xpui.js makes spotify slow/ not responsive (by directly editing xpui.js)
> But changing the only the xpui.css does not give that problem

Anyone else experiencing this? 
Works fine for me

**Comment by Nuzair46** [comment]
@voltrare @Araxeus No. The change in xpui.js will not impact performance as it is just a simple toggle rather than having a long function to do complex stuff.

**Comment by voltrare** [comment]
I understand, its just it happens like that for me even after reboot and changing back to default value returns the state (not laggy)

**Comment by ZeldrexK0DE** [comment]
> I have a workaround that works for now.
> 
> 1. Navigate to `%APPDATA%/Spotify/Apps`. Here you will find xpui.spa, make a copy of this file so that you have a backup.
> 2. Rename xpui.spa to xpui.zip and open the zip. Within the zip, there will be xpui.css, extract this file.
> 3. Open the extracted file with a text editor and search for `.b2731cdeb97c1f6a4cf5caa44e943acb-scss`. This is the css selector of the ad banner div.
> 4. For this selector, change the `display:flex` to `display:none`, save the file, **drop it back in the zip to replace** the original xpui.css.
> 5. Change the extension from xpui.zip to xpui.spa
> 
> The div selector did not change even after I reinstalled, so I'm guessing it should be the same for everybody.

This file is completely empty for me it's literally 22bytes

**Comment by voltrare** [comment]
@ZeldrexK0DE try to reinstall

**Comment by hyoretsu** [comment]
The latest .bat should close this issue, no? Though I advise adding a warning telling people to download it if they want to remove the empty banner. (I always wanted to, but I had no idea it also did that)

**Comment by jimmysofat6864** [comment]
> > I have a workaround that works for now.
> > 
> > 1. Navigate to `%APPDATA%/Spotify/Apps`. Here you will find xpui.spa, make a copy of this file so that you have a backup.
> > 2. Rename xpui.spa to xpui.zip and open the zip. Within the zip, there will be xpui.css, extract this file.
> > 3. Open the extracted file with a text editor and search for `.b2731cdeb97c1f6a4cf5caa44e943acb-scss`. This is the css selector of the ad banner div.
> > 4. For this selector, change the `display:flex` to `display:none`, save the file, **drop it back in the zip to replace** the original xpui.css.
> > 5. Change the extension from xpui.zip to xpui.spa
> > 
> > The div selector did not change even after I reinstalled, so I'm guessing it should be the same for everybody.
> 
> The new class is .snMc3rl8H2pM9cqkTpQ6. Just follow the same instructions but instead of .b2731cdeb97c1f6a4cf5caa44e943acb use the new class

I'm editing the xpui file on Mac so it should be the same thing but I don't know what class the upgrade button is called. Do you know what it's called? I can locate the `height:106px` so that's no problem. Just trying to see how I can remove the upgrade button.

**Comment by amitkuliyal** [comment]
latest version of Spotify is now showing adds, no update to this project from quite some time, this the project still alive?

**Comment by Araxeus** [comment]
> latest version of Spotify is now showing adds, no update to this project from quite some time, this the project still alive?

It works perfectly for me 😮

**Comment by Nuzair46** [comment]
Closing this as its almost 9 months old and a workaround is available with auto installer.

**Comment by hemtopo** [comment]
hi, the file was work excellent, but now I have the next error:
Invoke-WebRequest : 404: Not Found
En línea: 1 Carácter: 85
+ ... pe]::Tls12; Invoke-WebRequest -UseBasicParsing 'https://raw.githubuse ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest], WebExc
   eption
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand

thanks for your help

**Comment by hemtopo** [comment]
hi, the file was work excellent, but now I have the following error message:

![imagen](https://user-images.githubusercontent.com/102632607/181116108-6192dd04-2502-43ce-9cad-2b91e04426d1.png)

Invoke-WebRequest : 404: Not Found
En línea: 1 Carácter: 85
+ ... pe]::Tls12; Invoke-WebRequest -UseBasicParsing 'https://raw.githubuse ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest], WebExc
   eption
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand

Thanks for your help

**Comment by JuanBetto** [comment]
There is any solution for remove "Upgrade" bottom?

**Comment by NguyenASang** [comment]
@JuanBetto I tested on my Window 11 with Spotify ver 1.1.91.824.g07f1e963 and everything works perfectly (Upgrade button and Ads have been removed)

**Comment by JuanBetto** [comment]
> 

Yeah, but my friend activity is hidden :/, that's normal?

**Comment by JuanBetto** [comment]
> @JuanBetto I tested on my Window 11 with Spotify ver 1.1.91.824.g07f1e963 and everything works perfectly (Upgrade button and Ads have been removed)

Yeah, but my friend activity is hidden :/, that's normal?

**Comment by NguyenASang** [comment]
@JuanBetto 
I think you disabled this one

![image](https://user-images.githubusercontent.com/87893636/182022939-a750e5f3-366f-4d0c-839d-ebbead328be2.png)

To show "friend activity" you need to enable that option.
